src/backend/pipeline/stream_fdw.c

```diff
@@ -532,20 +513,44 @@ BeginStreamModify(ModifyTableState *mtstate, ResultRelInfo *result_info,
 			ack->batch = batch;
 		}
 
-		sis->worker_queue = get_any_worker_queue_with_lock();
+		/*
+		 * We always write to the same worker from a combiner process to prevent
+		 * unnecessary reordering
+		 */
+		if (IsContQueryCombinerProcess())
+		{
+			int idx = MyContQueryProc->group_id % continuous_query_num_workers;
+			sis->worker_queue = get_worker_queue_with_lock(idx, false);
+		}
+		else
+		{
```

Comment:
Technically the style guide says that braces should be used for either all `if` blocks or none :/